miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -863,20 +953,103 @@ export.get_note_serial_number
 end
 
 #! Tells the transaction kernel that we are about to execute a procedure on a foreign account.
-#!
-#! Inputs:  [KERNEL_PROCEDURE_HASH, foreign_account_id, pad(11)]
-#! Outputs: [pad(16)]
+#! Loads the data of the specified foreign account to the memory and moves the pointer of the 
+#! currently accessing account to it. 
```

Comment:
I would expand on this a bit. For example, we load account data only if it hasn't been already loaded, right?